api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityServiceTest.java

```diff
@@ -16,63 +25,140 @@
+        given(eligibilityStatusCalculator.determineEligibilityStatusForNewClaim(any())).willReturn(ELIGIBLE);
+        given(cycleEntitlementCalculator.calculateEntitlement(any(), any(), any())).willReturn(voucherEntitlement);
 
-        EligibilityResponse result = eligibilityService.determineEligibilityForNewClaimant(claimant);
+        EligibilityAndEntitlement result = eligibilityService.determineEligibilityForNewClaimant(claimant);
 
-        assertThat(result).isEqualTo(eligibilityResponse);
+        assertCorrectResponse(result, INELIGIBLE, eligibilityResponse, voucherEntitlement);
         verify(claimRepository).liveClaimExistsForNino(claimant.getNino());
         verify(client).checkEligibility(claimant);
-        verify(eligibilityStatusCalculator).determineEligibilityStatus(eligibilityResponse);
+        verify(eligibilityStatusCalculator).determineEligibilityStatusForNewClaim(eligibilityResponse);
+        verify(cycleEntitlementCalculator).calculateEntitlement(
+                Optional.ofNullable(claimant.getExpectedDeliveryDate()),
+                eligibilityResponse.getDateOfBirthOfChildren(),
+                LocalDate.now());
     }
 
     @Test
-    void shouldCheckEligibilityServiceForExistingClaimant() {
+    void shouldReturnEligibleForExistingClaimantWithVouchers() {
         Claimant claimant = aValidClaimant();
         EligibilityResponse eligibilityResponse = anEligibilityResponse();
+        LocalDate cycleStartDate = LocalDate.now().minusDays(1);
+        PaymentCycleVoucherEntitlement voucherEntitlement = aPaymentCycleVoucherEntitlementWithVouchers();
+        PaymentCycle previousCycle = PaymentCycleTestDataFactory.aValidPaymentCycle();
         given(client.checkEligibility(any())).willReturn(eligibilityResponse);
+        given(cycleEntitlementCalculator.calculateEntitlement(any(), any(), any(), any())).willReturn(voucherEntitlement);
 
-        EligibilityResponse result = eligibilityService.determineEligibilityForExistingClaimant(claimant);
+        EligibilityAndEntitlement result = eligibilityService.determineEligibilityForExistingClaimant(claimant, cycleStartDate, previousCycle);
 
-        assertThat(result).isEqualTo(eligibilityResponse);
+        assertCorrectResponse(result, eligibilityResponse.getEligibilityStatus(), eligibilityResponse, voucherEntitlement);
         verify(client).checkEligibility(claimant);
+        verify(cycleEntitlementCalculator).calculateEntitlement(
+                Optional.ofNullable(claimant.getExpectedDeliveryDate()),
+                eligibilityResponse.getDateOfBirthOfChildren(),
+                cycleStartDate,
+                previousCycle.getVoucherEntitlement());
         verifyZeroInteractions(eligibilityStatusCalculator, claimRepository);
     }
 
+    @Test
+    void shouldReturnIneligibleForExistingClaimantWithoutVouchers() {
+        Claimant claimant = aValidClaimant();
+        EligibilityResponse eligibilityResponse = anEligibilityResponse();
```

Comment:
Can we be specific about the `EligibilityStatus` here as its hard to read without specifically setting it here, and its important to the test that we know what it is.

Reply:
I did think about renaming `anEligiblityResponse` to `anEligibleResponse` - would that work? (It would affect a lot more classes though)